Realm/RLMObjectBase.mm

```diff
@@ -208,6 +210,16 @@ - (void)setValue:(id)value forUndefinedKey:(NSString *)key {
                                                              RLMSchema.partialPrivateSharedSchema)];
             }
         }
+        else if (property.set) {
```

Comment:
This could be shared between array and set pretty easily. Perhaps rename both _rlmArray and _rlmSet to _rlmCollection so that the same code works on both?

Reply:
renaming both to `_rlmCollection ` resulted in a compiler error. I managed to reduce the code with 
```
id collection = property.array ?
                [object_getIvar(self, ivar) _rlmArray] : [object_getIvar(self, ivar) _rlmSet];
```